Dockerfile.QA

```diff
@@ -70,6 +69,13 @@ RUN wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/nul
     apt-get install -y --no-install-recommends \
       cmake-data=3.21.1-0kitware1ubuntu20.04.1 cmake=3.21.1-0kitware1ubuntu20.04.1
 
+# Install boost version >= 1.78 for boost::span
+# Current libboost-dev apt packages are < 1.78, so install from tar.gz
+RUN wget -O ${TRITONTMP_DIR}/boost.tar.gz \
+        https://boostorg.jfrog.io/artifactory/main/release/1.80.0/source/boost_1_80_0.tar.gz && \
+    (cd ${TRITONTMP_DIR} && tar xzf boost.tar.gz) && \
+    mv ${TRITONTMP_DIR}/boost_1_80_0/boost /usr/include/boost
```

Comment:
I wonder why we installed boost in both QA and build container but not in the actual linux container? Is boost not needed in the actual linux container and is only required for QA and build

Reply:
We don't have `/usr/include/boost` in the server container, I imagine because it's bundled at compile time.

Looks like we have boost in the QA container because we actually do a couple builds, ex: building python backend for triton-shm-monitor test. However, this newer boost version isn't required for that build, so I think this change is not needed in QA contianer. I'm checking on that now.

Reply:
Python backend fetches boost as a part of its own CMake so I think it doesn't need to install it manually.